Tests/UnitTests/Tests/AsyncBlockTests.cpp

```diff
@@ -554,6 +554,72 @@ DEFINE_TEST_CLASS(AsyncBlockTests)
         VERIFY_QUEUE_EMPTY(queue);
     }
 
+    struct SimpleProviderData
+    {
+        std::vector<XAsyncOp> ops{};
+    };
+
+    static HRESULT CALLBACK SimpleProvider(XAsyncOp op, XAsyncProviderData const* data)
+    {
+        auto providerData{ static_cast<SimpleProviderData*>(data->context) };
+        providerData->ops.push_back(op);
+
+        switch (op)
+        {
+        case XAsyncOp::Begin:
+        {
+            return XAsyncSchedule(data->async, 2000);
+        }
+        case XAsyncOp::DoWork:
+        {
+            XAsyncComplete(data->async, S_OK, 0);
+            return S_OK;
+        }
+        case XAsyncOp::Cancel:
+        {
+            HRESULT hr = XTaskQueueTerminate(data->async->queue, false, nullptr, nullptr);
+            return hr;
+            //return S_OK;
+        }
+        default:
+        {
+            return S_OK;
+        }
+        }
+    }
+
+    DEFINE_TEST_CASE(VerifyCancellation2)
+    {
+        XAsyncBlock async = {};
+        HRESULT hrCallback = E_UNEXPECTED;
+
+        CompletionThunk cb([&](XAsyncBlock* async)
+        {
+            hrCallback = XAsyncGetStatus(async, false);
+        });
+
+        XTaskQueueHandle localQueue{};
+        VERIFY_SUCCEEDED(XTaskQueueCreate(XTaskQueueDispatchMode::ThreadPool, XTaskQueueDispatchMode::ThreadPool, &localQueue));        
+
+        async.context = &cb;
+        async.callback = CompletionThunk::Callback;
+        async.queue = localQueue;
+
+        SimpleProviderData data{};
+
+        VERIFY_SUCCEEDED(XAsyncBegin(&async, &data, nullptr, nullptr, SimpleProvider));
+        VERIFY_ARE_EQUAL(E_PENDING, XAsyncGetStatus(&async, false));
+        XAsyncCancel(&async);
+        HRESULT hr = XAsyncGetStatus(&async, true);
+        VERIFY_ARE_EQUAL(S_OK, hr);
+
+        VerifyHasOp(data.ops, XAsyncOp::Begin);
+        VerifyHasOp(data.ops, XAsyncOp::Cancel);
+        VerifyHasOp(data.ops, XAsyncOp::Cleanup);
+
+        XTaskQueueCloseHandle(localQueue);
+    }
```

Comment:
This can be removed.  Was just using this to better understand XAsyncCancel mechanics